src/modm/board/nucleo_f031k6/module.lb

```diff
@@ -29,7 +29,10 @@ def prepare(module, options):
 
 def build(env):
     env.outbasepath = "modm/src/modm/board"
-    env.substitutions = {"board_has_logger": True}
+    env.substitutions = {
+        "with_logger": True,
+        "with_assert": env.has_module(":architecture:assert")
+    }
```

Comment:
We really need to deduplicate at least all the nucleo boards. This is getting ridiculous.